cheeseshop/swift.py

```diff
@@ -155,5 +159,17 @@ def __init__(self, keystone_session, region_id, interface='public',
     async def create_tempurl(self, name, container=None):
         assert container is not None or self.container is not None
         container = container or self.container
-
-        # TODO:greghaynes Implement this
+        method = 'PUT'
+        proto, barf, host, version, auth_account = self.endpoint.url.split('/')
+        location = "/".join([version, auth_account, container, name])
+        duration_in_seconds = 60 * 60 * 2
+        expires = int(time() + duration_in_seconds)
+        hmac_body = '{0}\n{1}\n/{2}'.format(method, expires, location)
+        sig = hmac.new(bytes(self.temp_url_key, 'ascii'),
```

Comment:
we should probably convert temp_url_key to bytes at the edges - i.e. when we read it in from the config perform temp_url_key.encode() or w/e and pass it around as bytes rather than as a string.